### PR TITLE
Use root typescript version 

### PR DESCRIPTION
**Pull request by ismail-syed**
## Description

Related to https://github.com/Shopify/quilt/issues/857 - Makes it easier to review the upcoming larger TS refactor PR of this repo

This PR removes all the monorepo package level TS dependencies and defers the use of typescript to be consumed from the root level of the project. 

**Why?**
- We don't really need this with Typescript project references moving forward
- Managing independent TS version for each monorepo package doesn't scale

### CODE REVIEW

File: packages/admin-graphql-api-utilities/package.json [hunk 1]
```diff
@@ -22,9 +22,7 @@
     "url": "https://github.com/shopify/quilt/issues"
   },
   "homepage": "https://github.com/Shopify/quilt/blob/master/packages/admin-graphql-api-utilities/README.md",
-  "devDependencies": {
-    "typescript": "~3.2.1"
-  },
+  "devDependencies": {},
```
Comment (TheMallen) [inline packages/admin-graphql-api-utilities/package.json]: 🔥  these empty fields IMO

File: packages/react-router/package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "react-router-dom": "^5.0.1"
   },
   "devDependencies": {
-    "typescript": "~3.2.1"
```
Comment (michelleyschen) [inline packages/react-router/package.json]: do we need this in `devDependencies` when it is in `dependencies` as well?

File: packages/react-serialize/package.json [hunk 3]
```diff
@@ -29,8 +29,5 @@
   "peerDependencies": {
     "react": "^16.0.0"
   },
-  "devDependencies": {
```
Comment (michelleyschen) [inline packages/react-serialize/package.json]: damn I really gonna kill this packages

File: packages/react-router/package.json (hunk 2 above)
Comment (ismail-syed) [reply packages/react-router/package.json]: Oh snap, good catch. It was from a bad rebase. I'll check through the rest

**Comment by michelleyschen** [comment]
I am not sure if I am understanding this all correctly but is there any chance that we can do the same thing for `tslib` ? and should we?

**Comment by ismail-syed** [comment]
> react-router is the only package I have some concern about.

That was a bad rebase, it's fixed now. 

> ... is there any chance that we can do the same thing for tslib ? and should we?

`tslib` is a dependency whereas `typescript` is a dev dependency. Consumers still need to install the specified `tslib`

**Pull request merged**